java/rocksjni/ratelimiterjni.cc

```diff
@@ -22,3 +23,98 @@ jlong Java_org_rocksdb_GenericRateLimiterConfig_newRateLimiterHandle(
       static_cast<int64_t>(jrefill_period_micros),
       static_cast<int32_t>(jfairness)));
 }
+
+/*
+ * Class:     org_rocksdb_RateLimiter
+ * Method:    newRateLimiterHandle
+ * Signature: (JJI)J
+ */
+jlong Java_org_rocksdb_RateLimiter_newRateLimiterHandle(
+    JNIEnv* env, jclass jclazz, jlong jrate_bytes_per_second,
+    jlong jrefill_period_micros, jint jfairness) {
+  auto* rate_limiter = rocksdb::NewGenericRateLimiter(
+      static_cast<int64_t>(jrate_bytes_per_second),
+      static_cast<int64_t>(jrefill_period_micros),
+      static_cast<int32_t>(jfairness));
+
+  // TODO(AR) `ptr_sptr_rate_limiter` is never deleted :-(
```

Comment:
@yhchiang This is a general problem in several places in the Java API where the C++ API makes use of `shared_ptr` not just here. I cannot see a fix for this at all and I have consulted with a couple of really good C++ developers and they also agree with me that `shared_ptr` in a C++ API seems to be incompatible with creating 3rd-party language bindings (such as JNI or even C).

So I should think we have a similar issue in the Rocks C API, would you know if we do?

If we are to fix this `shared_ptr` issue, I think that could be a separate PR which does it across the Java API.